### PR TITLE
Bug-305250_Fix

### PR DESCRIPTION
**Pull request by chshrikh**
The key value for username and password in environment property is
changed

### CODE REVIEW

File: Tasks/RunPowerShellOnTargetMachines/RunPowerShellOnTargetMachines.ps1 [hunk 1]
```diff
@@ -27,9 +27,9 @@ $port = '5985'
 
 $resources = Get-EnvironmentResources -EnvironmentName $environmentName -ResourceFilter $machineNames -Connection $connection -ErrorAction Stop
 
-$machineUserName = Get-EnvironmentProperty -EnvironmentName $environmentName -Key "Username" -Connection $connection -ErrorAction Stop
+$machineUserName = Get-EnvironmentProperty -EnvironmentName $environmentName -Key "Microsoft-Vslabs-MG-Resource-Username" -Connection $connection -ErrorAction Stop
```
Comment (pankajkhanzode) [inline Tasks/RunPowerShellOnTargetMachines/RunPowerShellOnTargetMachines.ps1]: Same as above. I would like us to change to right approach.

File: Tasks/CopyFilesToMachines/CopyFilesToMachines.ps1 [hunk 2]
```diff
@@ -24,9 +24,9 @@ $connection = Get-VssConnection -TaskContext $distributedTaskContext
 
 $resources = Get-EnvironmentResources -EnvironmentName $environmentName -ResourceFilter $machineNames -Connection $connection -ErrorAction Stop
 
-$machineUserName = Get-EnvironmentProperty -EnvironmentName $environmentName -Key "Username" -Connection $connection -ErrorAction Stop
+$machineUserName = Get-EnvironmentProperty -EnvironmentName $environmentName -Key "Microsoft-Vslabs-MG-Resource-Username" -Connection $connection -ErrorAction Stop
```
Comment (pankajkhanzode) [inline Tasks/CopyFilesToMachines/CopyFilesToMachines.ps1]: This is still flaky and will break when machine level override is in place. Why don't you use the cmdlet and get environment object and then use env.resource.UserName and env.Resource.Password?

Comment (chshrikh) [inline Tasks/CopyFilesToMachines/CopyFilesToMachines.ps1]: this work I have done and validating , but to get unblock we need to push this changes .

Comment (chshrikh) [inline Tasks/CopyFilesToMachines/CopyFilesToMachines.ps1]: I agree with Pankaj , will do the required fix

File: Tasks/CopyFilesToMachines/CopyFilesToMachines.ps1 [hunk 3]
```diff
@@ -43,6 +39,11 @@ if($deployFilesInParallel -eq "false" -or  ( $resources.Count -eq 1 ) )
 		$resOperationId = Invoke-ResourceOperation -EnvironmentName $environmentName -ResourceName $machine -EnvironmentOperationId $envOperationId -Connection $connection -ErrorAction Stop
 		Write-Verbose "ResourceOperationId = $resOperationId" -Verbose
 		
+		$machineUserName = $resource.Username
```
Comment (pankajkhanzode) [inline Tasks/CopyFilesToMachines/CopyFilesToMachines.ps1]: Can you convert tabs to spaces in this file, it has really affected formatting of code. Looks untidy.

**Comment by pankajkhanzode** [comment]
Signing off. Please verify it end-to-end before commit.

File: Tasks/RunPowerShellOnTargetMachines/RunPowerShellOnTargetMachines.ps1 [hunk 4]
```diff
@@ -21,18 +21,25 @@ Write-Verbose "runPowershellInParallel = $runPowershellInParallel" -Verbose
 import-module "Microsoft.TeamFoundation.DistributedTask.Task.Common"
 import-module "Microsoft.TeamFoundation.DistributedTask.Task.DevTestLabs"
 
+function GetResourceCredentials
```
Comment (pankagar13) [inline Tasks/RunPowerShellOnTargetMachines/RunPowerShellOnTargetMachines.ps1]: can we write it like this : Get-ResourceCredentials

File: Tasks/RunPowerShellOnTargetMachines/RunPowerShellOnTargetMachines.ps1 [hunk 5]
```diff
@@ -83,6 +94,10 @@ else
 		
 		Write-Verbose "ResourceOperationId = $resOperationId" -Verbose
 		
+		Write-Verbose "Get resource credentials" -Verbose
+		
+		$credential = GetResourceCredentials -resource $resource
+
```
Comment (pankagar13) [inline Tasks/RunPowerShellOnTargetMachines/RunPowerShellOnTargetMachines.ps1]: signing off

File: Tasks/RunPowerShellOnTargetMachines/RunPowerShellOnTargetMachines.ps1 [hunk 6]
```diff
@@ -50,6 +57,10 @@ if($runPowershellInParallel -eq "false" -or  ( $resources.Count -eq 1 ) )
```
Comment (nipunja) [inline Tasks/RunPowerShellOnTargetMachines/RunPowerShellOnTargetMachines.ps1]: looks good

File: Tasks/RunPowerShellOnTargetMachines/RunPowerShellOnTargetMachines.ps1 [hunk 7]
```diff
@@ -21,18 +21,25 @@ Write-Verbose "runPowershellInParallel = $runPowershellInParallel" -Verbose
 import-module "Microsoft.TeamFoundation.DistributedTask.Task.Common"
 import-module "Microsoft.TeamFoundation.DistributedTask.Task.DevTestLabs"
 
+function GetResourceCredentials
+{
+	param([object]$resource)
+
```
Comment (nipunja) [inline Tasks/RunPowerShellOnTargetMachines/RunPowerShellOnTargetMachines.ps1]: Write a pester test for this function

File: Tasks/RunPowerShellOnTargetMachines/RunPowerShellOnTargetMachines.ps1 (hunk 4 above)
Comment (chshrikh) [inline Tasks/RunPowerShellOnTargetMachines/RunPowerShellOnTargetMachines.ps1]: sure will take care of it

File: Tasks/RunPowerShellOnTargetMachines/RunPowerShellOnTargetMachines.ps1 (hunk 7 above)
Comment (chshrikh) [inline Tasks/RunPowerShellOnTargetMachines/RunPowerShellOnTargetMachines.ps1]: will add it as part of my user story

**Pull request merged**